dcos/cosmos.py

```diff
@@ -0,0 +1,401 @@
+        :type endpoint: str
+        :return: http method type
+        :rtype: str
+        """
+        method = self._http_method.get(endpoint)
+
+        if method is not 'post' and method is not 'get':
+            raise DCOSException('Bad method type')
+
+        return method
+
+    def _get_header(self, endpoint, version, headers=None):
+        """
+        Given an cosmos endpoint, a version, and any extra header values,
+        gets the header that can be used to query cosmos at endpoint.
+        Any key in headers will appear in the final header. In effect the
+        user function can overwrite the default header.
+
+        :param endpoint: a cosmos endpoint, of the form 'x/y',
+        for example 'package/repo/add' or 'service/start'
+        :type endpoint: str
+        :param version: The version of the request
+        :type version: str
+        :param headers: extra keys for the header
+        :type headers: dict[str, str]
+        :return: a header that can be used to query cosmos at endpoint
+        :rtype: dict[str, str]
+        """
+        simple_header = {
+            'Content-Type': self._get_content_type(endpoint),
+            'Accept': self._get_accept(endpoint, version)
+        }
+        return remove_nones(merge_dict(simple_header, headers))
+
+    def _endpoint_exists(self, endpoint):
+        """
+
+        :param endpoint: a possible cosmos endpoint
+        :type endpoint: str
+        :return: true if endpoint is a valid cosmos endpoint,
+        false otherwise
+        :rtype: bool
+        """
+        return endpoint in self._request_versions
+
+    def _get_accept(self, endpoint, version):
+        """
+        Gets the value for the Accept header key for
+        the cosmos request at endpoint.
+
+        :param endpoint: a cosmos endpoint, of the form 'x/y',
+        for example 'package/repo/add' or 'service/start'
+        :type endpoint: str
+        :param version: The version of the request
+        :type version: str
+        :return: the value for the Accept header key for endpoint
+        :rtype: str
+        """
+        if (endpoint, version) in self._special_accepts:
+            return self._special_accepts[(endpoint, version)]
+        return format_cosmos_header_type(endpoint, version, 'response')
+
+    def _get_content_type(self, endpoint):
+        """
+        Gets the value for the Content-Type header key for
+        the cosmos request at endpoint.
+
+        :param endpoint: a cosmos endpoint, of the form 'x/y',
+        for example 'package/repo/add' or 'service/start'
+        :type endpoint: str
+        :return: the value for the Content-Type header key for endpoint
+        :rtype: str
+        """
+        version = 'v1'
+        if (endpoint, version) in self._special_content_types:
+            return self._special_content_types[(endpoint, version)]
+        return format_cosmos_header_type(endpoint, version, 'request')
```

Comment:
To avoid the `version = v1` you can:

```
next(
  (value for ((current_endpoint, version), value) in _special_content_type.items() if current_endpoint == endpoint),
  None
)
```
Same as in Scala:
```
_special_content_types.find(
  case ((current_endpoint, _), _) => current_endpoint == endpoint
).map(case (_, value) => value)
```